R/arc-select.R

```diff
@@ -107,29 +105,16 @@ arc_select <- function(
   }
 
   # handle fields and where clause if missing
-  fields <- fields %||% query[["outFields"]] %||% "*"
-
-  # if not missing fields collapse to scalar character
-  if (length(fields) > 1) {
-    # check if incorrect field names provided
-    x_fields <- x[["fields"]][["name"]]
-    nindex <- tolower(fields) %in% tolower(x_fields)
-
-    # handle the case where a field is being selected that
-    # is not one of the available fields in the feature layer
-    if (any(!nindex)) {
-      cli::cli_abort(
-        "Field{?s} not in {.arg x}: {.var {fields[!nindex]}}"
-      )
-    }
-    # collapse together
-    fields <- paste0(fields, collapse = ",")
-  }
+  fields <- fields %||% query[["outFields"]]
+
+  fields <- match_fields(
+    fields = fields,
+    values = c(x[["fields"]][["name"]], "")
```

Comment:
@elipousson this is the only change I made. This will allow users to return only geometry if they so choose.